### ISSUE TITLE
A crash seems related to Auto Publishing Folders

### ISSUE THREAD

**Issue by livid** [Planetable/Planet#117]
So I have two published folders and turned on Auto Publishing. Now the app cannot start because of this crash.

<img width="1109" alt="Screenshot 2022-11-20 at 3 12 35 PM" src="https://user-images.githubusercontent.com/11427/202932425-a595b78b-3209-4b41-924d-c23a32dc1752.png">

**Comment by kailuo** [comment]
Indeed, the problem occurs under macOS 12 if auto publishing was turned on.

Added a fix in branch: https://github.com/Planetable/Planet/tree/improved-published-folders-4, will apply for another pr after some more testing.
